packages/react/src/components/Copy/Copy.js

```diff
@@ -6,91 +6,101 @@
  */
 
 import PropTypes from 'prop-types';
-import React, { Component } from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
+import debounce from 'lodash/debounce';
 import classnames from 'classnames';
 import { settings } from 'carbon-components';
 
 const { prefix } = settings;
 
-export default class Copy extends Component {
-  static propTypes = {
-    /**
-     * Pass in content to be rendred in the underlying <button>
-     */
-    children: PropTypes.node,
-
-    /**
-     * Specify an optional className to be applied to the underlying <button>
-     */
-    className: PropTypes.string,
-
-    /**
-     * Specify the string that is displayed when the button is clicked and the
-     * content is copied
-     */
-    feedback: PropTypes.string,
-
-    /**
-     * Specify the time it takes for the feedback message to timeout
-     */
-    feedbackTimeout: PropTypes.number,
-
-    /**
-     * Specify an optional `onClick` handler that is called when the underlying
-     * <button> is clicked
-     */
-    onClick: PropTypes.func,
-  };
-
-  static defaultProps = {
-    feedback: 'Copied!',
-    feedbackTimeout: 2000,
-    onClick: () => {},
+export default function Copy({
+  children,
+  className,
+  feedback,
+  feedbackTimeout,
+  onClick,
+  ...other
+}) {
+  const [animation, setAnimation] = useState('');
+  const classNames = classnames(className, {
+    [`${prefix}--copy-btn--animating`]: animation,
+    [`${prefix}--copy-btn--${animation}`]: animation,
+  });
+  const handleFadeOut = useCallback(
+    debounce(() => {
+      setAnimation('fade-out');
+    }, feedbackTimeout),
+    [feedbackTimeout]
+  );
+  const handleClick = useCallback(
```

Comment:
Style-nit: don't use `useCallback` unless it's for identity or we can show we have a performance problem without it.

Reply:
`useCallback` is necessary here for referential equality. without it the tooltip will not fade out because the `debounce` code receives a new callback function reference (since component will rerender before the `handleFadeOut` method fires)

Reply:
Totally got that for `handleFadeOut`, is the usage of it inside of `handleClick` causing this, or is it something else?

Reply:
I originally replied in the wrong thread but yeah for both the click handler and fadeout handler it's for cleanup purposes